apps/bluetooth/app.js

```diff
@@ -28,37 +28,43 @@ module.exports = function (activity) {
   var deviceProps = null
   var agent = null
   var currentSkillName = 'bluetooth'
-
-  function setAppType (skillName, afterFunc) {
+  var future = Promise.resolve()
```

Comment:
The `future` variable could be scoped under `serializePromise`.

Reply:
+1, we should remove the functional side effect ASAP.